src/services/download/http.rs

```diff
@@ -70,22 +71,20 @@ async fn start_request(
     .await
 }
 
-pub async fn download_source(
-    source: Arc<HttpSourceConfig>,
-    download_path: SourceLocation,
+/// Downloads the given `url` to `destination`.
+///
+/// This will internally follow HTTP redirects, adding the given `headers` to each intermediate
+/// request. It will further apply an exponential backoff in case of download errors.
+pub async fn download_url(
+    url: Url,
+    headers: &BTreeMap<String, String>,
```

Comment:
Why `BTreeMap` instead of `HashMap`, do you care about order?

Reply:
`BTreeMap` is generally the choice unless you're dealing with incredibly large maps. In small cases it is not different to a sorted vector, and then it fans out very nicely without the overhead of hashing the values. Depending on the hashing function, that's usually slower than inserting into moderately sized BTreeMaps.

Reply:
TIL, thanks!